packages/react-components/src/components/MessageThread.tsx

```diff
@@ -300,16 +301,17 @@ const memoizeAllMessages = memoizeFnAll(
       | ((messageStatusIndicatorProps: MessageStatusIndicatorProps) => JSX.Element | null)
       | undefined,
     defaultStatusRenderer: (
+      message: ChatMessage,
       status: MessageStatus,
-      messageThreadReadCount: number,
+      readReceiptForEachSender: { [key: string]: { lastReadMessage: string; name: string } },
       messageThreadParticipantCount: number
     ) => JSX.Element,
     defaultChatMessageRenderer: (message: MessageProps) => JSX.Element,
     strings: MessageThreadStrings,
     _attached?: boolean | string,
     statusToRender?: MessageStatus,
-    messageThreadReadCount?: number,
     messageThreadParticipantCount?: number,
+    readReceiptForEachSender?: { [key: string]: { lastReadMessage: string; name: string } },
```

Comment:
I am thinking of whether ForEachSender suffix is necessary or not... The type {string: {name, lastMessage}} indicates it is for sender ids, maybe name it directly as readReceipt or readReceiptByUserId (sender either means sender of the message or sender of the readReceipt, which is more confusing)

Also the property "name" is a little confusing, I would like to make it displayName so people know it is for user not something else

Reply:
Sorry readReceiptsBySenderId

Reply:
@prprabhu-ms @JamesBurnside as genius of naming, do you have any ideas of better name of this one?

Reply:
Yep I like
```
readReceipts: ReadReceiptsBySenderId
```

Reply:
There is another variable named readreceipts in the selectors, how about naming it to readReceiptsBySenderId instead?